lerobot/common/policies/vla/modeling_vla.py

```diff
@@ -0,0 +1,308 @@
+        q = k = self.maybe_add_pos_embed(x, decoder_pos_embed)
+        breakpoint()
+
+        # Self-attention
+        x = self.self_attn(q, k, value=x)[0]  # select just the output, not attention weights
+        x = skip + self.dropout1(x)
+
+        if self.pre_norm:
+            skip = x
+            x = self.norm2(x)
+        else:
+            x = self.norm1(x)
+            skip = x
+
+        # Cross-attention with encoder outputs
+        x = self.cross_attn(
+            query=self.maybe_add_pos_embed(x, decoder_pos_embed),
+            key=self.maybe_add_pos_embed(encoder_out, encoder_pos_embed),
+            value=encoder_out,
+        )[0]  # select just the output, not attention weights
+        x = skip + self.dropout2(x)
+
+        if self.pre_norm:
+            skip = x
+            x = self.norm3(x)
+        else:
+            x = self.norm2(x)
+            skip = x
+
+        # Feed-forward network
+        x = self.linear2(self.dropout(self.activation(self.linear1(x))))
+        x = skip + self.dropout3(x)
+
+        if not self.pre_norm:
+            x = self.norm3(x)
+
+        return x
+
+
+class ActionDecoder(nn.Module):
+    def __init__(self, config: VLAConfig):
+        """Runs multiple decoder layers followed by normalization."""
+        super().__init__()
+        self.layers = nn.ModuleList([ActionDecoderLayer(config) for _ in range(config.num_decoder_layers)])
+        self.norm = nn.LayerNorm(config.hidden_size)
+
+    def forward(
+        self,
+        x: torch.Tensor,
+        encoder_out: torch.Tensor,
+        decoder_pos_embed: torch.Tensor | None = None,
+        encoder_pos_embed: torch.Tensor | None = None,
+    ) -> torch.Tensor:
+        for layer in self.layers:
+            x = layer(
+                x, encoder_out, decoder_pos_embed=decoder_pos_embed, encoder_pos_embed=encoder_pos_embed
+            )
+        x = self.norm(x)
+        return x
+
+
+
+class VLA(nn.Module):
+    def __init__(self, config: VLAConfig):
+        super().__init__()
+
+        # Initialize the Qwen2VLForConditionalGeneration and ActionDecoder
+        #qwen2_vl_config = make_qwen2_vl_config(config)
+        self.chunk_size = config.chunk_size
+        self.decoder_pos_embed = nn.Embedding(config.chunk_size, config.hidden_size)
+        self.action_decoder = ActionDecoder(config)  # Use the updated ActionDecoder
+        self.action_head = nn.Linear(config.hidden_size, config.output_shapes["action"][0])
+        
+       
+        self.half()
+
+    def forward(self, hidden_states):
+            """
+            Forward pass to compute action logits using hidden states from Qwen2VL (Llava).
+            
+            Args:
+            hidden_states: Tensor of shape [batch_size, seq_len, hidden_size] from Llava model.
+            
+            Returns:
+            action_logits: Tensor of predicted actions.
+            """
+            batch_size = hidden_states.size(0)  # Ensure batch size is extracted
```

Comment:
This part is the core and I think it might contain the mistakes. the way we repeat the input to match the chunk_size and also the encoder_out value needs to be checked and reviewed.